hw/mcu/dialog/da1469x/src/da1469x_clock.c

```diff
@@ -342,29 +385,36 @@ da1469x_clock_calibrate(uint8_t clock_sel, uint16_t ref_cnt)
     assert(!(ANAMISC_BIF->CLK_REF_SEL_REG & ANAMISC_BIF_CLK_REF_SEL_REG_REF_CAL_START_Msk));
 
     ANAMISC_BIF->CLK_REF_CNT_REG = ref_cnt;
-    ANAMISC_BIF->CLK_REF_SEL_REG = (clock_sel << ANAMISC_BIF_CLK_REF_SEL_REG_REF_CLK_SEL_Pos) |
-                                   ANAMISC_BIF_CLK_REF_SEL_REG_REF_CAL_START_Msk;
 
+    /* Select reference clock & calibrated clock */
+    ANAMISC_BIF->CLK_REF_SEL_REG =
+        (DA1469X_REF_SEL << ANAMISC_BIF_CLK_REF_SEL_REG_CAL_CLK_SEL_Pos) |
+        (clock_sel << ANAMISC_BIF_CLK_REF_SEL_REG_REF_CLK_SEL_Pos);
+
+    /* Start measurement */
+    ANAMISC_BIF->CLK_REF_SEL_REG |= ANAMISC_BIF_CLK_REF_SEL_REG_REF_CAL_START_Msk;
+
+    /* Wait for meaurement to complete */
     while (ANAMISC_BIF->CLK_REF_SEL_REG & ANAMISC_BIF_CLK_REF_SEL_REG_REF_CAL_START_Msk);
 
     ref_val = ANAMISC_BIF->CLK_REF_VAL_REG;
 
     da1469x_pd_release(MCU_PD_DOMAIN_PER);
 
-    return 32000000 * ref_cnt / ref_val;
+    return CAL_REF_FREQ * ref_cnt / ref_val;
 }
 
 void
 da1469x_clock_sys_rc32m_calibrate(void)
 {
-    g_mcu_clock_rc32m_freq = da1469x_clock_calibrate(1, 100);
+    g_mcu_clock_rc32m_freq = da1469x_clock_calibrate(DA1469X_CALIB_RC32M, RC32M_CAL_REF_CNT);
```

Comment:
ref cnt should be configurable via syscfg for consistency

Reply:
Done.